api/namex/utils/queue_util.py

```diff
@@ -0,0 +1,50 @@
+import json
+import time
+import uuid
+from datetime import datetime, timezone
+
+from config import Config
+from namex.services import queue
+
+
+def create_cloud_event_msg(msg_id, msg_type, source, time, identifier, json_data_body):  # pylint: disable=too-many-arguments # noqa E501
+    # industry standard arguments for this message
+    """Creates a payload for the cloud event messages."""
+    cloud_event_msg = {
+        'specversion': '1.x-wip',
+        'type': msg_type,
+        'source': source,
+        'id': msg_id,
+        'time': time,
+        'datacontenttype': 'application/json',
+        'identifier': identifier
+    }
+    if json_data_body:
+        cloud_event_msg['data'] = json_data_body
+
+    return cloud_event_msg
+
+def create_name_request_state_msg(nr_num, state_cd, old_state_cd):
+    """Builds a name request state message."""
+    return create_cloud_event_msg(msg_id=str(uuid.uuid4()),
+                                            msg_type='bc.registry.names.request',
+                                            source=f'/requests/{nr_num}',
```

Comment:
the source typically provides information on where the message originated from.  `/namex_api/request/{nr_num}` is more ideal i think.

Reply:
This change can come from different requests because it is triggered by the model listener.

Reply:
right, good call